src/coreComponents/physicsSolvers/contact/SolidMechanicsAugmentedLagrangianContact.cpp

```diff
@@ -0,0 +1,1347 @@
+          {
+            for( localIndex j=0; j < dofColIndicesDisp.size(); ++j )
+            {
+              pattern.insertNonZero( eqnRowIndicesBubble[i], dofColIndicesDisp[j] );
+            }
+          }
+        }
+
+      }
+
+    } );
+
+    SurfaceElementRegion const & region = elemManager.getRegion< SurfaceElementRegion >( getUniqueFractureRegionName() );
+    FaceElementSubRegion const & subRegion = region.getUniqueSubRegion< FaceElementSubRegion >();
+    ArrayOfArraysView< localIndex const > const elemsToFaces = subRegion.faceList().toViewConst();
+    ArrayOfArraysView< localIndex const > const faceToNodeMap = faceManager.nodeList().toViewConst();
+
+    static constexpr int maxNumDispFaceDof = 3 * 4;
+
+    for( localIndex kfe=0; kfe<subRegion.size(); ++kfe )
+    {
+
+      localIndex const numNodesPerFace = faceToNodeMap.sizeOfArray( kfe );
+      localIndex const numDispDof = 3*numNodesPerFace;
+
+      for( int k=0; k<2; ++k )
+      {
+        localIndex const kf = elemsToFaces[kfe][k];
+        localIndex const kf_other = elemsToFaces[kfe][(1+k)%2];
+
+        // working arrays
+        stackArray1d< globalIndex, maxNumDispFaceDof > eqnRowIndicesDisp ( numDispDof );
+        stackArray1d< globalIndex, 3 > eqnRowIndicesBubble( 3 );
+        stackArray1d< globalIndex, maxNumDispFaceDof > dofColIndicesDisp ( numDispDof );
+        stackArray1d< globalIndex, 3 > dofColIndicesBubble( 3 );
+
+        for( localIndex idof = 0; idof < 3; ++idof )
+        {
+          eqnRowIndicesBubble[idof] = bubbleDofNumber[kf] + idof - rankOffset;
+          dofColIndicesBubble[idof] = bubbleDofNumber[kf] + idof;
+        }
+
+        for( localIndex a=0; a<numNodesPerFace; ++a )
+        {
+          const localIndex & node = faceToNodeMap( kf_other, a );
+          for( localIndex idof = 0; idof < 3; ++idof )
+          {
+            eqnRowIndicesDisp[3*a + idof] = dispDofNumber[node] + idof - rankOffset;
+            dofColIndicesDisp[3*a + idof] = dispDofNumber[node] + idof;
+          }
+        }
+
+        for( localIndex i = 0; i < eqnRowIndicesDisp.size(); ++i )
+        {
+          if( eqnRowIndicesDisp[i] >= 0 && eqnRowIndicesDisp[i] < pattern.numRows() )
+          {
+            for( localIndex j = 0; j < dofColIndicesBubble.size(); ++j )
+            {
+              pattern.insertNonZero( eqnRowIndicesDisp[i], dofColIndicesBubble[j] );
+            }
+          }
+        }
+
+        for( localIndex i = 0; i < eqnRowIndicesBubble.size(); ++i )
+        {
+          if( eqnRowIndicesBubble[i] >= 0 && eqnRowIndicesBubble[i] < pattern.numRows() )
+          {
+            for( localIndex j=0; j < dofColIndicesDisp.size(); ++j )
+            {
+              pattern.insertNonZero( eqnRowIndicesBubble[i], dofColIndicesDisp[j] );
+            }
+          }
+        }
+
+      }
+    }
+  } );
+
+}
+
+// TODO: Is it possible to define this method once? Similar to SolidMechanicsLagrangeContact
```

Comment:
what do you mean?

Reply:
This function is very similar to the one used in `SolidMechanicsLagrangeContact`.
I was wondering if a new method could be defined in ContactSolverBase, for example. 
Anyway I will have a clearer idea about this function when the slip mode for ALM is also implemented.
In the meantime may I leave this comment as a reminder?